### PR TITLE
customize CSS namespace

### PR DESCRIPTION
**Pull request by giladgray**
support customizing the `pt-` namespace on all classes.

new `$pt-namespace` (aliased as `$ns` for brevity) Sass variable determines prefix for all generated CSS classes.
use environment variable `process.env.BLUEPRINT_NAMESPACE` to customize prefix in JS `classes.ts` constants files.
docs CSS examples can safely use `#{$ns}` in modifier lists and it will be replaced at runtime.

TODO

- [x] hardcoded (non-`Classes`) strings in JS code
- [x] hardcoded string in MD docs 😢 
- [x] KSS styleguide IDs
- [x] documentation

### CODE REVIEW

File: packages/core/src/common/_variables.scss [hunk 1]
```diff
@@ -4,6 +4,9 @@
 @import "colors";
 @import "mixins";
 
+$pt-namespace: "pt" !default;
```
Comment (llorca) [inline packages/core/src/common/_variables.scss]: why not one var like `$namespace: "pt" !default` or `$blueprint-namespace: "pt" !default`?

File: packages/core/src/common/classes.ts [hunk 2]
```diff
@@ -7,207 +7,239 @@
 import { Alignment } from "./alignment";
 import { Intent } from "./intent";
 
+const NS = "bp3";
```
Comment (JKillian) [inline packages/core/src/common/classes.ts]: this namespace is bp3 but the one above is pt?

Comment (giladgray) [reply packages/core/src/common/classes.ts]: yeah oops shhhhhh

**Comment by giladgray** [comment]
gonna pull out some changes from this PR into separate ones to shrink the diff.

**Pull request merged**